wiki/App-Initialization-Flow.md

```diff
@@ -0,0 +1,68 @@
+4. `initAppState()` (`state/index.ts`) receives a `customDomain` argument from `useInitApp()`, and hits our `/status` endpoint.
+    1. The `app` state variable is defined within `state/index.ts`, and its values are updated by the `initAppState`, using the `data` object returned by the `/status` route.
+        - More broadly, `initAppState()` is called on:app initialization, user login, user logout, and chain community creation.
+    2. When the `/status` fetch is finished, app `isLoading` state is set to false.
+5. The `/status` endpoint is divided into two main data-fetches, `getChainStatus()` and `getUserStatus()`. `getChainStatus()` is always called; `getUserStatus()` is only called if a `user` object has been passed with the request.
+    1. `getChainStatus()`: Grabs ChainStore, ChainNodes, NotificationCategories, and CommunitySnapshotSpaces, and threadCountQueryData entries.
+    2. `getUserStatus`: Grabs user associations (e.g. selected chain instance, social accounts, addresses). Once addresses have been grabbed, various rows associated with those addresses (e.g. discussion drafts, roles) are requested in turn.
+6. Once the `/status` data is received on callback, `initAppState()` clears and repopulates app state.
+    1. Relevant controllers (e.g. user, config) are cleared.
+    2. Nodes, recent activity, snapshot chains, roles, chain categories, notification categories, and notification category types are all repopulated. 
+    3. Active user and login state are updated.
+        - The websocket connection is initialized or disconnected depending on whether the user is logged in or out, respectively.
+        - Sets the users’ starred communities, selected chain, and display name
+    4. If a custom domain is passed as an argument to the invocation of `initAppState`, the `setCustomDomain` setter is fired. If the function's `shouldRedraw` param is set to true, then the app's `loginStateEmitter` emits a redraw.
+7. A `Router` component within the `App` component conditionally renders if state variable `isLoading` is false; this router uses `react-router-dom` library methods to conditionally create either custom domain routes, or common domain routes, depending on whether a custom domain param is passed.
+    1. All routes are configured in the `client/scripts/navigation` directory.
+    2. `CommonDomainRoutes.tsx` and `CustomDomainRoutes.tsx` declare the routes to front-end pages, switched in `navigation/Router.tsx` based on results of the `/domain` call.
+    3. "General routes," referring to pages like `/privacy` and `/terms`, are also loaded via `GeneralRoutes.tsx`.
+8. Most of these routes are set to return their relevant views wrapped inside our `LayoutWrapper` component, via the function `withLayout()`, imported from `client/scripts/views/Layout.tsx`. 
+    1. The `LayoutWrapper` injects the page’s specific params into the page as child, and injects the `scope` and `deferChain` arguments into the `Layout` component.
```

Comment:
Should remove the deferChain reference here @gdjohnson

Reply:
Good catch—done.